pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go

```diff
@@ -121,17 +122,33 @@ func (c *ClusterRoleAggregationController) syncClusterRole(key string) error {
 		return nil
 	}
 
-	// we need to update
-	clusterRole := sharedClusterRole.DeepCopy()
-	clusterRole.Rules = nil
-	for _, rule := range newPolicyRules {
-		clusterRole.Rules = append(clusterRole.Rules, *rule.DeepCopy())
-	}
-	_, err = c.clusterRoleClient.ClusterRoles().Update(context.TODO(), clusterRole, metav1.UpdateOptions{})
+	clusterRole := rbacv1ac.ClusterRole(sharedClusterRole.Name).
+		WithRules(toApplyPolicyRules(newPolicyRules)...)
+
+	opts := metav1.ApplyOptions{FieldManager: "clusterrole-aggregation-controller", Force: true}
+	_, err = c.clusterRoleClient.ClusterRoles().Apply(context.TODO(), clusterRole, opts)
```

Comment:
does this gracefully fall back to the current logic if apply is not enabled on the target server (e.g. the beta feature is disabled)?

Reply:
No, good catch. Having a look.

Reply:
Added the fallback to Update and comments explaining.  Also added disabled SSA cases to the tests.

Sanity check that HTTP 415 (Unsupported Media Type) is the right response code:

```
git diff hack/local-up-cluster.sh

-FEATURE_GATES=${FEATURE_GATES:-"AllAlpha=false"}
+FEATURE_GATES=${FEATURE_GATES:-"AllAlpha=false,ServerSideApply=false"}

./hack/local-up-cluster.sh
…

cluster/kubectl.sh apply -f hack/testdata/pod.yaml --server-side
error: Server-side apply not available on the server: (415: Unsupported Media Type)
```